src/hotspot/share/gc/g1/g1BlockOffsetTable.inline.hpp

```diff
@@ -113,9 +113,13 @@ inline size_t G1BlockOffsetTablePart::block_size(const HeapWord* p) const {
 }
 
 inline HeapWord* G1BlockOffsetTablePart::block_at_or_preceding(const void* addr) const {
-  assert(_object_can_span || _bot->offset_array(_bot->index_for(_hr->bottom())) == 0,
-         "Object crossed region boundary, found offset %u instead of 0",
-         (uint) _bot->offset_array(_bot->index_for(_hr->bottom())));
+#ifdef ASSERT
+  if (!_hr->is_continues_humongous()) {
+    // For non ContinuesHumongous regions, the first obj always starts from bottom.
+    u_char offset = _bot->offset_array(_bot->index_for(_hr->bottom()));
+    assert(offset == 0, "Found offset %u instead of 0", offset);
```

Comment:
Please add region# too here, maybe even the region's short type string (that can be gained from the `hs_err` file though).